### ISSUE TITLE
[code cleanup] Split BrowserViewController.swift into multiple files

### ISSUE THREAD

**Issue by garvankeeley** [mozilla-mobile/firefox-ios#1941]
This file is an overly long monolith of Space Odyssey proportions, consider splitting up by extension type for the larger extensions, like so:

```
mkdir Client/Frontend/Browser/BrowserViewController
```

and in there maybe 6-8 files along with the main BVC.swift with larger extensions pulled out:

```
BrowserViewController+ReaderMode.swift
BrowserViewController+ToolbarDelegate.swift
BrowserViewController+URLBarDelegate.swift (or combine these two, just giving examples)
BrowserViewController+ etc ...
```

If no interest, feel free to close, if interest don't mind doing.

**Comment by varkor** [comment]
I think this is something people have been talking about doing for a while now, but noöne's quite got round to it. I'm sure a pull request would be appreciated!

**Comment by skyylex** [comment]
Hello @garvankeeley, @varkor.

I want to help with this task, however, I think not only splitting, but also refactoring would be nice to apply. My question is how it would be better to organize such process? I don't think it's good idea to perform work as a single transaction. From my experience it's better (if possible) to use atomic changes with small controllable impact to the related source code. Please share your thoughts.

**Comment by garvankeeley** [comment]
Perhaps someone from the project can comment, but I think for a big action like this, there needs to be a bugzilla bug instead of github-only. 

If you look at https://github.com/brave/browser-ios/tree/master/Client/Frontend/Browser/BrowserViewController I did the BVC split there, but the codebase is diverged from this one. 
These are the categories I followed:
BrowserViewController+BrowserDelegate.swift 	
	BrowserViewController+ContextMenuHelperDelegate.swift 	
	BrowserViewController+FindInPage.swift 
	BrowserViewController+KeyCommands.swift 
	BrowserViewController+ReaderModeDelegate.swift
	BrowserViewController+TabManagerDelegate.swift 
	BrowserViewController+ToolbarAndUrlbarDelegate.swift 
	BrowserViewController+WKNavigationDelegate.swift 
	BrowserViewController+WKUIDelegate.swift

Yes to splitting up commits because that (nearly) always makes reviews easier (when I don't it is because I am too lazy to do the extra work required for that).
